src/main/osd/osd_elements.c

```diff
@@ -1962,7 +2017,7 @@ static void osdDrawSingleElement(displayPort_t *osdDisplayPort, uint8_t item)
     element.attr = DISPLAYPORT_SEVERITY_NORMAL;
 
     // Call the element drawing function
-    if ((item >= OSD_SYS_GOGGLE_VOLTAGE) && (item < OSD_ITEM_COUNT)) {
+    if ((item >= OSD_SYS_GOGGLE_VOLTAGE) && (item <= OSD_SYS_FAN_SPEED)) {
         displaySys(osdDisplayPort, elemPosX, elemPosY, (displayPortSystemElement_e)(item - OSD_SYS_GOGGLE_VOLTAGE + DISPLAYPORT_SYS_GOGGLE_VOLTAGE));
```

Comment:
@ledvinap @SpencerGraffunder Right now the problem is that the `OSD_SYS_x` elements are blocking other elements from getting added to the end of the enumeration, I reckon.

![grafik](https://github.com/betaflight/betaflight/assets/19867640/8ce1cd02-5823-40c3-995d-c64d047a4ae2)

So I'd suggest changing the `osd_items_e` enums like this to always put `OSD_SYS_x` elements last:

```c
typedef enum {
    ...

    OSD_GPS_LAP_TIME_CURRENT,
    OSD_GPS_LAP_TIME_PREVIOUS,
    OSD_GPS_LAP_TIME_BEST3,
    OSD_ITEM_COUNT_VISIBLE,  // MUST BE LAST (visible item)

    // OSD_SYS_XXX items are rendered in the goggles firmware
    OSD_SYS_GOGGLE_VOLTAGE = OSD_ITEM_COUNT_VISIBLE,
    OSD_SYS_VTX_VOLTAGE,
    OSD_SYS_BITRATE,
    OSD_SYS_DELAY,
    OSD_SYS_DISTANCE,
    OSD_SYS_LQ,
    OSD_SYS_GOGGLE_DVR,
    OSD_SYS_VTX_DVR,
    OSD_SYS_WARNINGS,
    OSD_SYS_VTX_TEMP,
    OSD_SYS_FAN_SPEED,
    OSD_ITEM_COUNT  // MUST BE LAST (sys item)
} osd_items_e;
```

Then we can solve this a bit more cleanly and we don't need to touch it anymore when adding new osd items:

```suggestion
    if ((item >= OSD_ITEM_COUNT_VISIBLE) && (item < OSD_ITEM_COUNT)) {
        displaySys(osdDisplayPort, elemPosX, elemPosY, (displayPortSystemElement_e)(item - OSD_ITEM_COUNT_VISIBLE + DISPLAYPORT_SYS_GOGGLE_VOLTAGE));
```

Reply:
We wouldn’t have to change those lines at all after this if new elements keep getting added to the end like they’re supposed to. I assume shifting the sys elements down every time an element is added would also require more configurator integration. Correct me if I’m wrong though. Maybe the solution would be just a comment that explains what’s going on there. There’s nothing majorly wrong with it, it’s just confusing if you don’t already know because it’s not self explanatory.

Reply:
Yeah, I have not thought about the configurator part. You are probably right. Forget about this suggestion.